src/bindings/c/include/openvino/c/ov_infer_request.h

```diff
@@ -0,0 +1,167 @@
+    void* args;
+} ov_callback_t;
+
+/**
+ * @struct ov_ProfilingInfo_t
+ * @brief Store profiling info data
+ */
+typedef struct {
+    enum Status {       //!< Defines the general status of a node.
+        NOT_RUN,        //!< A node is not executed.
+        OPTIMIZED_OUT,  //!< A node is optimized out during graph optimization phase.
+        EXECUTED        //!< A node is executed.
+    } status;
+    int64_t real_time;      //!< The absolute time, in microseconds, that the node ran (in total).
+    int64_t cpu_time;       //!< The net host CPU time that the node ran.
+    const char* node_name;  //!< Name of a node.
+    const char* exec_type;  //!< Execution type of a unit.
+    const char* node_type;  //!< Node type.
+} ov_profiling_info_t;
+
+/**
+ * @struct ov_profiling_info_list_t
+ * @brief A list of profiling info data
+ */
+typedef struct {
+    ov_profiling_info_t* profiling_infos;
+    size_t size;
+} ov_profiling_info_list_t;
+
+// infer_request
+/**
+ * @defgroup infer_request infer_request
+ * @ingroup openvino_c
+ * Set of functions representing of infer_request.
+ * @{
+ */
+
+/**
+ * @brief Sets an input/output tensor to infer on.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param tensor_name  Name of the input or output tensor.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_tensor(ov_infer_request_t* infer_request, const char* tensor_name, const ov_tensor_t* tensor);
+
+/**
+ * @brief Sets an input tensor to infer on.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param idx Index of the input tensor. If @p idx is greater than the number of model inputs, an exception is thrown.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_input_tensor(ov_infer_request_t* infer_request, size_t idx, const ov_tensor_t* tensor);
+
+/**
+ * @brief Gets an input/output tensor to infer on.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param tensor_name  Name of the input or output tensor.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_get_tensor(const ov_infer_request_t* infer_request, const char* tensor_name, ov_tensor_t** tensor);
+
+/**
+ * @brief Gets an output tensor to infer on.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param idx Index of the tensor to get.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_get_output_tensor(const ov_infer_request_t* infer_request, size_t idx, ov_tensor_t** tensor);
```

Comment:
what's about set output tensor by index

Reply:
Actually, we didn't wrap all C++ api in the first version, because it will be a huge number api from C++ to C mapping. We just map the APIs that are called in C sample so that can create a basic C inference task.